lint_test/avoid_unused_parameters_test.dart

```diff
@@ -0,0 +1,109 @@
+// ignore_for_file: prefer_const_declarations
+// ignore_for_file: unnecessary_nullable_for_final_variable_declarations
+// ignore_for_file: unused_local_variable
+// ignore_for_file: unused_element
+// ignore_for_file: newline-before-return
+
+import 'package:flutter/material.dart';
+
+/// Check the `avoid-unused-parameters` rule
+
+// expect_lint: avoid-unused-parameters
+void fun(String s) {
+  return;
+}
+
+class TestClass {
+  // expect_lint: avoid-unused-parameters
+  static void staticMethod(int a) {}
+
+  // expect_lint: avoid-unused-parameters
+  void method(String s) {
+    return;
+  }
+
+  void methodWithUnderscores(int _) {}
+}
+
+class TestClass2 {
+  void method(String _) {
+    return;
+  }
+}
+
+class SomeOtherClass {
+  // expect_lint: avoid-unused-parameters
+  void method(String s) {
+    return;
+  }
+}
+
+class SomeAnotherClass extends SomeOtherClass {
+  @override
+  void method(String s) {}
+}
+
+void someOtherFunction(String s) {
+  print(s);
+  return;
+}
+
+class SomeOtherAnotherClass {
+  void method(String s) {
+    print(s);
+    return;
+  }
+
+  // expect_lint: avoid-unused-parameters
+  void anonymousCallback(Function(int a) cb) {}
+}
+
+// expect_lint: avoid-unused-parameters
+void closure(int a) {
+  void internal(int a) {
+    print(a);
+  }
+}
+
+typedef MaxFun = int Function(int a, int b);
+
+// Allowed same way as override
+final MaxFun maxFunInstance = (int a, int b) => 1;
```

Comment:
@DenisBogatirov What's your take on this?

Reply:
Didn't quite get the question.

Type is `MaxFun` so we are obliged to provide given set of params, if we remove one then there would be a type mismatch. That would work same way with with name params because if we remove `namedParam` it wouldn't be a defined type.

Or do you want to create another test case for named params with typedef?

Reply:
With positional params, we have the freedom to create a placeholder arg with the correct type. Just need to match the signature.
Here's how I see it:

```dart
// expect_lint: avoid-unused-parameters
final MaxFun maxFunLint = (int a, int b) => 1;
  
final MaxFun maxFunOk = (int _, int __) => 1;
```

Reply:
Original rule does not care about parameter name as long as function or method is implemented.

Should I implement this logic? It will affect `@override` as well

Reply:
Hm, is there no way to know that we're operating on a typedef instead of a class method override?

Reply:
At least at this time I couldn't find any :(

Reply:
Well, in that case let's document this in an issue and get back to it later on.
You've done a great job expanding the rule to constructors already.